tests/test_ranges.py

```diff
@@ -1124,6 +1124,26 @@ def test_union_of_overlapping_ranges(self):
                 )
             )
 
+        @pytest.mark.xfail
```

Comment:
xfail here since in this commit the test will fail - the xfail is removed later in the PR.